File: 剑指39.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        if not nums:
            return 0
        hashset = {}
        for num in nums:
            hashset[num] = hashset[num] + 1 if hashset.get(num) else 1
        for key, value in hashset.items():
            if value > len(nums) / 2:
                return key
        return 0
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        nums.sort()
        return nums[len(nums) // 2]

File: 剑指39.java
class Solution {   //哈希表
    public int majorityElement(int[] nums) {
        int bound = nums.length / 2;
        Map<Integer, Integer> counts = new HashMap<>();
        for(int num: nums){
            if(!counts.containsKey(num)) {
                counts.put(num, 1);
            }else{
                counts.put(num, counts.get(num) + 1);
            }
            if (counts.get(num) > nums.length / 2) return num;
        }
        return 0;
    }
}

class Solution {  // 排序
    public int majorityElement(int[] nums) {
        Arrays.sort(nums);
        return nums[nums.length / 2];
    }
}
